ci_tools/process_results.py

```diff
@@ -55,14 +55,22 @@
             print(code, msg)
             if level == 'failure':
                 file, start, end = get_code_file_and_lines(file_name, pyccel_folder = pyccel_folder)
-                if not code.startswith('GL'):
+                # if code isn't "The object does not have a docstring"
+                if code != 'GL08':
+                    # Restrict error reporting line to docstring
                     with open(os.path.join(pyccel_folder, file), 'r', encoding='utf-8') as code_file:
                         lines = code_file.readlines()[start-1:end+2]
                     lines = [l.strip() for l in lines]
-                    doc_openings = [i for i,l in enumerate(lines) if l.startswith('"""') or l.endswith('"""')]
+                    doc_openings = []
+                    for i,l in enumerate(lines):
+                        if l.startswith('"""'):
+                            doc_openings.append(i)
+                        if l != '"""' and l.endswith('"""'):
+                            doc_openings.append(i)
```

Comment:
The new code looks **very** similar to the original one. The only case where it behaves differently is when the docstring is on a single line, in which case you append the line twice now. I do not quite understand why you want this.

Reply:
I need to have a pair of indices for each docstring. One for the start line and one for the end line.
Occasionally people write docstrings like:
```
def f():
    """ Tiny docstring """
```
With the old code this causes problems as the start is identified, but the end is either the next docstring or isn't found. You then get an `IndexError` when you try to access it